### DISCUSSION TITLE
Error updating workflow: Request entity too large

### DISCUSSION THREAD

**Discussion by yichenche** [argoproj/argo-workflows#7257] [Q&A] [answered]
I was trying to use argo workflow to run with a workflow containing 10k parallel steps (though set the parallelism to 240). But the workflow stuck around finishing 6000/10K pods, and got the following error when looking at the logs in  workflow controller:

level=warning msg="Error updating workflow: Request entity too large: limit is 3145728 RequestEntityTooLarge" namespace=default workflow=6378fbe9-0edc-4cb9-9165-bd3a7d14dc52

I am using an AWS managed EKS cluster, and using argo 3.2.4, with default executor type set to emissary. I just contacted AWS support, EKS does not support changing  -max-request-bytes in etcd.

I would be grateful if anyone can give suggestion here how to workaround this issue.

**Comment by thegreymatter** [comment, marked as answer]
You need to enable nodeStatusOffLoad.
https://argoproj.github.io/argo-workflows/offloading-large-workflows/

In a nutshell, Argo by default saves the status of the DAG inside a K8S resource. for smaller DAG's it's fine. for a larger DAG it starts to hit the limit of K8S itself. that is why there is the ability to allow Argo to store the status in persistent DB instead.
